test/mocks/providers/provider_ranking.dart

```diff
@@ -1,40 +1,50 @@
 import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:firebase_core/firebase_core.dart";
 import "package:flutter/material.dart";
 import "package:hooks_riverpod/hooks_riverpod.dart";
 import "package:proxima/services/database/firestore_service.dart";
 import "package:proxima/viewmodels/login_view_model.dart";
 import "package:proxima/views/pages/home/content/ranking/ranking_page.dart";
 
 import "../data/firestore_user.dart";
+import "../overrides/override_user_centauri_points_view_model.dart";
+import "../services/setup_firebase_mocks.dart";
 
 /// Returns a testing [ProviderContainer] which purpose is to test the ranking
 /// view model.
 /// **Note**: it also sets the [testingUserFirestore] on the provided database.
 Future<ProviderContainer> rankingProviderContainerWithTestingUser(
   FakeFirebaseFirestore fakeFirebaseFirestore,
 ) async {
+  setupFirebaseAuthMocks();
+  await Firebase.initializeApp();
   await setUserFirestore(fakeFirebaseFirestore, testingUserFirestore);
 
   return ProviderContainer(
     overrides: [
       firestoreProvider.overrideWithValue(fakeFirebaseFirestore),
       validLoggedInUserIdProvider.overrideWithValue(testingUserFirestoreId),
+      ...mockUserCentauriPointsViewModelCentauriOverride(
+        centauriPoints: testingUserData.centauriPoints,
+      ),
     ],
   );
 }
 
 /// Returns a testable [RankingPage] already overriden by providers
 /// of [rankingProviderContainerWithTestingUser].
-Future<MaterialApp> rankingPageMockApp(
+Future<ProviderScope> rankingPageMockApp(
   FakeFirebaseFirestore fakeFirebaseFirestore,
 ) async {
   final rankingContainer =
       await rankingProviderContainerWithTestingUser(fakeFirebaseFirestore);
 
-  return MaterialApp(
-    home: UncontrolledProviderScope(
-      container: rankingContainer,
-      child: const RankingPage(),
+  return ProviderScope(
```

Comment:
By humble curiosity, why did you wrapped this `MaterialApp` in a `ProviderScope`?

Reply:
A test was failing on a pop-up that was not finding the `ProviderScope`.

Thanks to you, I came back on this, and realised that we already had a `ProviderScope` but that was not wrapping the `MaterialApp` as we usually do, that is why I wrapped it.

Fixed in: [6b09986](https://github.com/ProximaEPFL/proxima/pull/348/commits/6b0998610759ac85c5dc49153e8ebe78e2f1ed05)

Thank you 💯

Reply:
Ah yes, I understand better now! Great, thank you 🚀